Makefile

```diff
@@ -0,0 +1,390 @@
+IMAGE_RUST             = cryptocom/chain
+IMAGE_TENDERMINT       = tendermint/tendermint:v0.32.8
+DOCKER_FILE            = docker/Dockerfile.chain
+DOCKER_FILE_RELEASE    = docker/Dockerfile.release
+ITEMS_START            = sgx-validation sgx-query chain-abci tendermint client-rpc
+ITEMS_STOP             = client-rpc tendermint chain-abci sgx-query sgx-validation
+
+create-path:
+	mkdir -p ${HOME}/.cargo/{git,registry}
+	bash -c "mkdir -p $(data_path)/tendermint/{config,data}"
+	bash -c "mkdir -p $(data_path)/{wallet,chain-storage,enclave-storage}"
+
+init-tendermint:
+ifeq ($(chain), devnet)
+	@echo "\033[32mcopy devnet tendermint config\033[0m"
+	cp docker/config/devnet/tendermint/config.toml $(data_path)/tendermint/config/
+	cp docker/config/devnet/tendermint/genesis.json $(data_path)/tendermint/config/
+	cp docker/config/devnet/tendermint/priv_validator_key.json $(data_path)/tendermint/config/
+	cp docker/config/devnet/tendermint/priv_validator_state.json $(data_path)/tendermint/data/
+else ifeq ($(chain), testnet)
+	@echo "\033[32mcopy testnet tendermint config\033[0m"
+	bash -c "cp docker/config/testnet/tendermint/{config.toml,genesis.json} $(data_path)/tendermint/config/"
+else ifeq ($(chain), mainnet)
+	@echo "\033[32mcopy mainnet tendermint config\033[0m"
+	bash -c "cp docker/config/mainnet/tendermint/{config.toml,genesis.json} $(data_path)/tendermint/config/"
+endif
+
+install-sgx-driver:
+ifeq ($(SGX_MODE), HW)
+	@if [ -e "/dev/isgx" ] || [ -e "/dev/sgx" ]; then \
+		echo "\033[32msgx driver already installed\033[0m"; \
+	else \
+		echo "\033[32install sgx driver\033[0m"; \
+		curl --proto '=https' -sSf https://download.01.org/intel-sgx/sgx-linux/2.7.1/distro/ubuntu18.04-server/sgx_linux_x64_driver_2.6.0_4f5bb63.bin > /tmp/driver.bin && \
+		chmod +x /tmp/driver.bin &&\
+		sudo /tmp/driver.bin && \
+		rm /tmp/driver.bin; \
+	fi
+else
+	@echo "\033[32mSGX_MODE is SW, no need to install sgx driver\033[0m"
+endif
+
+# build the sgx image
+image:
+ifeq ($(DOWNLOAD_URL)X, X)
+	@echo "\033[32mbuild docker image with local binary\033[0m";
+	chmod +x ci-scripts/*.sh;
+	docker build -t $(IMAGE):$(TAG) -f $(DOCKER_FILE)  --build-arg BUILD_MODE=$(build_mode) .
+else
+	@echo "\033[32mdownload binary and build docker image\033[0m";
+	chmod +x docker/*.sh;
+	docker build -t $(IMAGE):$(TAG) -f $(DOCKER_FILE_RELEASE) --build-arg DOWNLOAD_URL=$(DOWNLOAD_URL) .
+endif
+
+# build the chain binary in docker
+build-chain:
+	docker run -i --rm \
+		-v ${HOME}/.cargo/git:/root/.cargo/git \
+		-v ${HOME}/.cargo/registry:/root/.cargo/registry \
+		-v `pwd`:/chain \
+		--env RUSTFLAGS=-Ctarget-feature=+aes,+sse2,+sse4.1,+ssse3 \
+		--workdir=/chain \
+		$(IMAGE_RUST):latest \
+		bash -c ". /root/.docker_bashrc && \
+		$(CARGO_BUILD_CMD) --manifest-path=dev-utils/Cargo.toml && \
+		cd chain-abci && $(CARGO_BUILD_CMD_ABCI) && \
+		cd ../client-cli && $(CARGO_BUILD_CMD_CLI) && \
+		cd ../client-rpc && $(CARGO_BUILD_CMD_CLI)"
```

Comment:
I think the build commands caused the problem, although all the commands run successfully,  some binary disappeared in the `target` directory.  I'll have a more test

Reply:
you may need extra rules in dockerignore, as suggested by @calvinaco https://github.com/crypto-com/chain/pull/780#discussion_r365864560